### PR TITLE
makefiles/docker.inc.mk: add and use generic docker_run_make

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

Split from #12419 this PR adds a generic `docker_run_make` function to run make commands in docker.

- `DOCKER_MAKE_ARGS` is now empty since `DOCKER_MAKECMDGOALS` is passed individually yo the function and `DOCKER_OVERRIDE_CMDLINE` is allays passed.
- `BUILD_DIR` creation is removed since it is provided globally.

The function 3rd argument is currently unused, but I use it in #12419 to set `flags` specific to some docker targets, notably `flash` and `term`.

### Testing procedure

- Green murdock

- Run docker targets and verify it builds in docker

- compile with and without `DOCKER_MAKE_ARGS="-j"`

- any ideas on what else to test?

### Issues/PRs references

Split from #12419

### CODE REVIEW

File: makefiles/docker.inc.mk [hunk 1]
```diff
@@ -91,8 +90,15 @@ DOCKER_OVERRIDE_CMDLINE += $(strip $(DOCKER_OVERRIDE_CMDLINE_AUTO))
...
-DOCKER_MAKE_ARGS += $(DOCKER_MAKECMDGOALS) $(DOCKER_OVERRIDE_CMDLINE)
+# Set default run flags:
+# - allocate a pseudo-tty
+# - remove container on exit
+# - set username/UID to executor
+DOCKER_USER ?= $$(id -u)
+DOCKER_RUN_FLAGS ?= --rm --tty --user $(DOCKER_USER)
+
+# allow setting make args from command line like '-j'
+DOCKER_TARGET_MAKE_ARGS ?=
```
Comment (aabadie) [inline makefiles/docker.inc.mk:101]:
```suggestion
DOCKER_MAKE_ARGS ?=
```

**Comment by aabadie** [comment]
@fjmolinas I like this PR. Please address my comment and we are good.

File: makefiles/docker.inc.mk (hunk 1 above)
Comment (fjmolinas) [reply makefiles/docker.inc.mk]: done

**Comment by aabadie** [comment]
All green, Go!

**Comment by fjmolinas** [comment]
Thanks for the review @aabadie!

**Pull request merged**